File: 2-NORMAL/Python/7-validate-parentheses/proposed-solution/main.py
# Challenge: Create a function that, given a string composed of parentheses and possibly other opening and closing symbols, verifies if the sequence is valid (i.e., each open symbol has its corresponding closing symbol in the correct order). Use a stack data structure to solve it.

from parentheses import is_valid_parentheses

def main():
    test_strings = ["()", "()[]{}", "(]", "([)]", "{[]}"]
    for s in test_strings:
        print(f"Is '{s}' valid? {is_valid_parentheses(s)}")

if __name__ == "__main__":
    main()

File: 2-NORMAL/JavaScript/7-validate-parentheses/proposed-solution/main.js
 // main.js - Example usage of ParenthesesValidator class

import ParenthesesValidator from './parentheses.js';

/*
Challenge: Create a function that, given a string composed of parentheses and possibly other opening and closing symbols, verifies if the sequence is valid (i.e., each open symbol has its corresponding closing symbol in the correct order). Use a stack data structure to solve it.
*/

function main() {
  const testStrings = ["()", "()[]{}", "(]", "([)]", "{[]}"];
  testStrings.forEach(s => {
    console.log(`Is '${s}' valid?`, ParenthesesValidator.isValidParentheses(s));
  });
}

main();
